contrib/src/main/scala/akka/stream/contrib/Valve.scala

```diff
@@ -50,27 +50,30 @@ object Valve {
 }
 
 /**
- * Materializes into a [[ValveSwitch]] which provides a the method flip that stops or restarts the flow of elements passing through the stage. As long as the valve is closed it will backpressure.
+ * Materializes into a [[Future]] of [[ValveSwitch]] which provides a the method flip that stops or restarts the flow of elements passing through the stage. As long as the valve is closed it will backpressure.
  *
  * Note that closing the valve could result in one element being buffered inside the stage, and if the stream completes or fails while being closed, that element may be lost.
  *
  * @param mode state of the valve at the startup of the flow (by default Open)
  */
-final class Valve[A](mode: SwitchMode) extends GraphStageWithMaterializedValue[FlowShape[A, A], ValveSwitch] {
+final class Valve[A](mode: SwitchMode) extends GraphStageWithMaterializedValue[FlowShape[A, A], Future[ValveSwitch]] {
 
   val in: Inlet[A] = Inlet[A]("valve.in")
 
   val out: Outlet[A] = Outlet[A]("valve.out")
 
   override val shape = FlowShape(in, out)
 
-  override def createLogicAndMaterializedValue(inheritedAttributes: Attributes): (GraphStageLogic, ValveSwitch) = {
+  override def createLogicAndMaterializedValue(inheritedAttributes: Attributes): (GraphStageLogic, Future[ValveSwitch]) = {
     val logic = new ValveGraphStageLogic(shape, mode)
-    (logic, logic.switch)
+    (logic, logic.promise.future)
   }
 
   private class ValveGraphStageLogic(shape: Shape, var mode: SwitchMode) extends GraphStageLogic(shape) with InHandler with OutHandler {
-    val switch = new ValveSwitch {
+
+    val promise = Promise[ValveSwitch]
```

Comment:
ok, I guess this is the same as issue https://github.com/akka/akka/issues/20503
`Future` is not so nice, but we can't do much else until that ticket is fixed